lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +30,115 @@ const _changesRequiredOutput = """
+    VersionConstraint.parse('^5.0.0')
+  ],
+  'over_react': [
+    VersionConstraint.parse('^2.0.0'),
+    VersionConstraint.parse('^3.0.0')
+  ]
+};
+
+VersionConstraint getDependencyVersion(String packageName) {
+  try {
+    if (pubspecYaml.containsKey('dependencies') &&
+        pubspecYaml['dependencies'].containsKey(packageName)) {
+      return VersionConstraint.parse(pubspecYaml['dependencies'][packageName]);
+    } else if (pubspecYaml.containsKey('dev_dependencies') &&
+        pubspecYaml['dev_dependencies'].containsKey(packageName)) {
+      return VersionConstraint.parse(
+          pubspecYaml['dev_dependencies'][packageName]);
+    }
+  } catch (e) {
+    return null;
+  }
+  return null;
+}
+
 void main(List<String> args) {
-  final query = FileQuery.dir(
-    pathFilter: isDartFile,
-    recursive: true,
-  );
-  exitCode = runInteractiveCodemodSequence(
-    query,
-    [
-      ReactDomRenderMigrator(),
-      ReactStyleMapsUpdater(),
-    ],
-    args: args,
-    defaultYes: true,
-    changesRequiredOutput: _changesRequiredOutput,
-  );
-
-  final logger = Logger('over_react_codemod.fixmes');
-  for (var dartFile in query.generateFilePaths()) {
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  Map<String, bool> inTransition = {};
+  bool foundReactOrOverReact = false;
+
+  final react16CodemodLogger = Logger('over_react_codemod.react16');
+  react16CodemodLogger.onRecord.listen((LogRecord record) {
+    print(record.message);
+  });
+  react16CodemodLogger
+      .info('Checking if project needs to run $react16CodemodName...');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e) {
+    if (e is FileSystemException) {
+      react16CodemodLogger
+          .warning('Could not find pubspec.yaml, exiting codemod.');
+    } else if (e is YamlException || e is ArgumentError) {
+      react16CodemodLogger
+          .warning('pubspec.yaml is unable to be parsed, exiting codemod.');
+    }
+    exitCode = 0;
+  }
+  for (var package in packagesToCheckFor.entries) {
+    var constraint = getDependencyVersion(package.key);
+    if (constraint != null) {
+      foundReactOrOverReact = true;
+      react16CodemodLogger
+          .info('Found ${package.key} with version ${constraint}');
+      // Found it so lets add it to the in transition list to false until its
+      // validated that we know it is in transtition.
+      inTransition[package.key] = false;
+    }
+    if (constraint != null &&
+        !constraint.isAny &&
+        constraint.allowsAny(package.value[0]) &&
+        constraint.allowsAny(package.value[1])) {
+      inTransition[package.key] = true;
+    }
+  }
+  if (foundReactOrOverReact) {
+    react16CodemodLogger.info(inTransition);
```

Comment:
This line in the output isn't super helpful:
> Checking if project needs to run React16 Codemod...
> Found react with version ^4.4.2
> {react: false}
> pubspec.yaml does not have transition versions of react or over_react, exiting codemod.
> We are all done here, Byeee!

Can we prefix it with a label (e.g., `In transition: $inTransition`) or remove it?